providers/aws-ssm/pkg/provider.go

```diff
@@ -0,0 +1,55 @@
+package awsssm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-feature/go-sdk-contrib/providers/aws-ssm/pkg/service"
+	"github.com/open-feature/go-sdk/openfeature"
+)
+
+
+type Provider struct {
+	svc *service.AWS
+}
+
+func NewProvider() (*Provider, error) {
+	svc, err := service.NewAWSService()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize AWS SSM provider: %w", err)
+	}
+
+	return &Provider{
+		svc: svc,
+	}, nil
+}
+
+func (p *Provider) Metadata() openfeature.Metadata {
+	return openfeature.Metadata{
+		Name: "AWS System Manager Provider",
+	}
+}
+
+func (p *Provider) Hooks() []openfeature.Hook {
+	return []openfeature.Hook{}
+}
+
+func (p Provider) BooleanEvaluation(ctx context.Context, flag string, defaultValue bool, evalCtx openfeature.FlattenedContext) openfeature.BoolResolutionDetail {
```

Comment:
```suggestion
func (p Provider) BooleanEvaluation(ctx context.Context, flag string, defaultValue bool, flatCtx openfeature.FlattenedContext) openfeature.BoolResolutionDetail {
```
Same suggestion for the other evaluation functions, to more accurately reflect the parameter type.